pool/workers.go

```diff
@@ -0,0 +1,139 @@
+package pool
```

Comment:
There is a lot of state management involved here. I think it might be worth to wrap it with a struct(s).

Something like this (it's not complete code, just an idea)
```
type Pool struct {
  maxSize uint // limit of workers
  currentSize uint // global counter of workers
  jobs map[connection.ID]job // maps from connectionId -> job
  logger *zap.SugaredLogger
}

func New (size uint, events <-chan *watcher.Event, logger) *Pool {}
func (pool *Pool) Run() {}
func (pool *Pool) Close() {}
```

and `job` type would be following struct 

```
type job struct {
  workers []workers
  connection *connection.Connection
  shutdown <-chan // used to stop processing in every worker
}
```

I think it would be beneficial for readability and managing the pool to have all workers processing one connection under one object. 

What do you think?